core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -71,6 +72,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
   // Executors we have requested the cluster manager to kill that have not died yet
   private val executorsPendingToRemove = new HashSet[String]
 
+  /**
+   * Send new credentials to executors. This is the method that is called when the scheduled
+   * login completes, so the new credentials can be sent to the executors.
+   * @param credentials
+   */
+  def sendNewCredentialsToExecutors(credentials: SerializableBuffer): Unit = {
+    // We don't care about the reply, so going to deadLetters is fine.
```

Comment:
No, that is not necessarily true. The initial startup would basically happen with credentials from the startup (similar to how it is now) -- the tokens are set up by YARN. YARN handles the NM tokens but not the HDFS tokens, which need to be replaced. This patch handles that, so it is ok to start running - since the initial tokens will last it a while.